plugins/defaultplugins/ifplugin/interface_config.go

```diff
@@ -295,6 +296,96 @@ func (plugin *InterfaceConfigurator) configRxMode(iface *intf.Interfaces_Interfa
 	return err
 }
 
+func (plugin *InterfaceConfigurator) configureIPAddresses(ifName string, ifIdx uint32, addresses []*net.IPNet, unnumbered *intf.Interfaces_Interface_Unnumbered) error {
+	if unnumbered != nil && unnumbered.IsUnnumbered {
+		ifNameIP := unnumbered.InterfaceWithIP
+		if ifNameIP == "" {
+			return fmt.Errorf("unnubered interface %s has no interface with IP address set", ifName)
+		}
+		ifIdxIP, _, found := plugin.swIfIndexes.LookupIdx(ifNameIP)
+		if !found {
+			// cache not-configurable interface
+			plugin.unnumberedIfaces[ifName] = ifNameIP
+			plugin.Log.Debugf("unnubered interface %s requires IP address from non-existing %v, moved to cache", ifName, ifNameIP)
+			return nil
+		}
+		// Set interface as un-numbered
+		err := vppcalls.SetUnnumberedIP(ifIdx, ifIdxIP, plugin.Log, plugin.vppCh, measure.GetTimeLog(interfaces.SwInterfaceSetUnnumbered{}, plugin.Stopwatch))
+		if err != nil {
+			return err
+		}
+		// just log
+		if len(addresses) != 0 {
+			plugin.Log.Warnf("Interface %v set as un-numbered contains IP address(es)", ifName, addresses)
+		}
+	}
+
+	// configure optional ip address
+	var wasErr error
+	for i := range addresses {
```

Comment:
why not
for _, address := range addresses
?

Reply:
changed to keep the same look with other usages